<a href="https://colab.research.google.com/github/kamilediasdeoliveira-ui/Estrutura_de_Dados_II/blob/main/aula_03_estrutura_de_dados_ii.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# EXERCÍCIO 1 - REFERÊNCIAS

a = [10, 20, 30]
b = a

b.append(40)

print("a:",a)
print("b:",b)
print("id(a):", id(a))
print("id(b):", id(b))


a: [10, 20, 30, 40]
b: [10, 20, 30, 40]
id(a): 131946976211776
id(b): 131946976211776


'\n\n\n'

EXERCÍCIO 1

1. Qual será a saída?
R = a: [10, 20, 30, 40]
b: [10, 20, 30, 40]
id(a): 131946976211776
id(b): 131946976211776
\n\n\n

2. a e b representam o mesmo objeto?
R = Sim, eles referenciam o mesmo objeto

3. Por quê?
R = Por que a lista origina é modificada sem fazer uma nova lista e b passa a ser uma referência para a

4. O que aconteceria com b = a.copy()?
R = a saída de a seria 10, 20, 30
e a saída de b 10,20,30, 40
por que cria uma cópia da lista original e depois adiciona o novo elemento

In [ ]:
# EXERCÍCIO 2 - CONSTRUINDO UMA CADEIA

